File: src-tauri/src/api/vod.rs
use serde::{Deserialize, Serialize};
use anyhow::Result;
use super::client::{XtreamClient, deserialize_option_string};
use super::common::CategoryApi;

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct VodStreamApi {
    pub stream_id: i64,
    pub name: Option<String>,
    pub stream_icon: Option<String>,
    #[serde(default, deserialize_with = "deserialize_option_string")]
    pub category_id: Option<String>,
    #[serde(default, deserialize_with = "deserialize_option_string")]
    pub rating: Option<String>,
    pub container_extension: Option<String>,
    #[serde(default, deserialize_with = "deserialize_option_string")]
    pub added: Option<String>,
    #[serde(default)]
    pub is_favorite: Option<i32>,
    #[serde(default)]
    pub profile_id: Option<i64>,
}

pub async fn fetch_categories(client: &XtreamClient) -> Result<Vec<CategoryApi>> {
    client.fetch(Some("get_vod_categories")).await
}

pub async fn fetch_streams(client: &XtreamClient) -> Result<Vec<VodStreamApi>> {
    client.fetch(Some("get_vod_streams")).await
}

pub async fn fetch_vod_info(client: &XtreamClient, stream_id: i64) -> Result<serde_json::Value> {
    let action = format!("get_vod_info&vod_id={}", stream_id);
    client.fetch(Some(&action)).await
}

File: src/types/vod.ts
export interface VodCategory {
  profile_id: number
  category_id: string
  category_name: string
}

export interface VodStream {
  profile_id: number
  stream_id: number
  name: string | null
  stream_icon: string | null
  category_id: string | null
  rating: string | null
  container_extension: string | null
  added: string | null
  is_favorite?: number
}

export interface VodInfo {
  profile_id: number
  stream_id: number
  info_json: string
  fetched_at: string
}
